### PR TITLE
[3435] Extract diagram style from useDropNode

### LINKED ISSUE #3435: Extract diagram style from useDropNode
useDropNode now will only export onNodeDragStart, onNodeDrag and onNodeDragStop,

### PR DESCRIPTION
**Pull request by mcharfadi**
Bug: https://github.com/eclipse-sirius/sirius-web/issues/3435

# Pull request template

## General purpose
What is the main goal of this pull request?
- [x] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/DiagramRenderer.tsx [hunk 1]
```diff
@@ -72,15 +72,16 @@ import { useShiftSelection } from './selection/useShiftSelection';
 import { useSnapToGrid } from './snap-to-grid/useSnapToGrid';
 
 import 'reactflow/dist/style.css';
+import { useDropDiagramStyle } from './dropNode/useDropNodeStyle';
```
Comment (frouene) [inline packages/diagrams/frontend/sirius-components-diagrams/src/renderer/DiagramRenderer.tsx]: wrong import order

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/DiagramRenderer.tsx [hunk 2]
```diff
@@ -72,15 +72,16 @@ import { useShiftSelection } from './selection/useShiftSelection';
...
 
-export const DiagramRenderer = ({ diagramRefreshedEventPayload }: DiagramRendererProps) => {
+export const DiagramRenderer = memo(({ diagramRefreshedEventPayload }: DiagramRendererProps) => {
   const { readOnly } = useContext<DiagramContextValue>(DiagramContext);
   const { diagramDescription } = useDiagramDescription();
   const { getEdges, onEdgesChange, getNodes, setEdges, setNodes } = useStore();
-  const nodes = getNodes();
-  const edges = getEdges();
+  const nodes = useMemo(() => getNodes(), [getNodes()]);
+  const edges = useMemo(() => getEdges(), [getEdges()]);
```
Comment (frouene) [inline packages/diagrams/frontend/sirius-components-diagrams/src/renderer/DiagramRenderer.tsx]: `getNodes() and getEdges()` return an array, I don't think using an array in deps is working

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/dropNode/useDropNode.ts [hunk 3]
```diff
@@ -191,69 +181,79 @@ export const useDropNode = (): UseDropNodeValue => {
         })
       );
     },
-    [reactFlowInstance.setNodes]
+    [reactFlowInstance]
```
Comment (frouene) [inline packages/diagrams/frontend/sirius-components-diagrams/src/renderer/dropNode/useDropNode.ts]: I'm not sure about using an object in deps

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/dropNode/useDropNodeStyle.ts [hunk 4]
```diff
@@ -43,3 +46,22 @@ export const useDropNodeStyle = (
 
   return { style };
 };
+
+export const useDropDiagramStyle = (): useDropDiagramStyleValue => {
```
Comment (frouene) [inline packages/diagrams/frontend/sirius-components-diagrams/src/renderer/dropNode/useDropNodeStyle.ts]: We should continue to have just one hook per file.
Please extract this one in a new file useDropDiagramStyle.ts

**Review by frouene — CHANGES_REQUESTED** [review]

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/dropNode/useDropNode.ts [hunk 5]
```diff
@@ -127,21 +126,14 @@ const useDropNodeMutation = () => {
...
-    setTargetNodeId,
-    resetDrop,
-  } = useContext<DropNodeContextValue>(DropNodeContext);
+  const { droppableOnDiagram, initialPosition, initialPositionAbsolute, initializeDrop, resetDrop } =
+    useContext<DropNodeContextValue>(DropNodeContext);
 
   const { diagramDescription } = useDiagramDescription();
   const reactFlowInstance = useReactFlow<NodeData, EdgeData>();
   const onDropNode = useDropNodeMutation();
   const { getNodes, getIntersectingNodes } = useReactFlow<NodeData, EdgeData>();
```
Comment (frouene) [inline packages/diagrams/frontend/sirius-components-diagrams/src/renderer/dropNode/useDropNode.ts]: Why declaring two times useReactFlow ?

File: packages/diagrams/frontend/sirius-components-diagrams/src/renderer/layout-events/useLayoutOnBoundsChange.ts [hunk 6]
```diff
@@ -27,7 +27,7 @@ export const useLayoutOnBoundsChange = (refreshEventPayloadId: string): UseLayou
   const { layout } = useLayout();
   const { synchronizeLayoutData } = useSynchronizeLayoutData();
   const { getEdges, getNodes } = useStore();
-  const { droppableOnDiagram } = useContext<DropNodeContextValue>(DropNodeContext);
+  const {} = useContext<DropNodeContextValue>(DropNodeContext);
```
Comment (frouene) [inline packages/diagrams/frontend/sirius-components-diagrams/src/renderer/layout-events/useLayoutOnBoundsChange.ts]: Is this necessary?

Comment (mcharfadi) [reply packages/diagrams/frontend/sirius-components-diagrams/src/renderer/layout-events/useLayoutOnBoundsChange.ts]: No, I forgot to remove this line

**Review by frouene — APPROVED** [review]

**Pull request merged**